app/js/components/maintenance/maintenanceCategorySelect/maintenanceCategorySelect.directive.js

```diff
@@ -50,8 +54,17 @@ function maintenanceCategorySelectCtrl($rootScope,
         return option.tag;
     }
 
+    function setUrl() {
+        if ($scope.selected.id !== 'all') {
+            $location.search('maintenanceCategory', $scope.selected.id);
+        } else if ($location.search().maintenanceCategory) {
+            delete $location.search().maintenanceCategory;
+        }
+    }
+
     $scope.$watch('category', function (category) {
         $scope.selected = (find($scope.options, {id: category}));
+        setUrl();
         $rootScope.$broadcast(Events.filters.tag,
                              getTag($scope.selected),
                              Events.filters.maintenanceCategorySelect);
```

Comment:
I think you can get rid of this category watch. Double check behind me but I don't think anything else is modifying this.

Reply:
@Kinlaw maintenance.controller.js has `$scope.setCategory` function that changes it